src/GLideNHQ/TxCache.cpp

```diff
@@ -854,6 +861,269 @@ bool TxFileStorage::isCached(Checksum checksum)
 	return _storage.find(checksum) != _storage.end();
 }
 
+/************************** TxNoCache *************************************/
+class TxNoCache : public TxCacheImpl
+{
+public:
+	TxNoCache(uint32 _options, const wchar_t *texturesPath, const wchar_t *ident, TxHiResCache *txHiResCache, dispInfoFuncExt callback);
+	~TxNoCache();
+
+	bool add(Checksum checksum, GHQTexInfo* info, int dataSize = 0) override { return true; }
+	bool get(Checksum checksum, GHQTexInfo *info) override;
+
+	bool save(const wchar_t* path, const wchar_t* filename, const int config) override { return true; }
+	bool load(const wchar_t* path, const wchar_t* filename, const int config, bool force) override { return true; }
+	bool del(Checksum checksum) override { return false; }
+	bool isCached(Checksum checksum) override { return true; }
+	void clear() override;
+	bool useReload() const override { return true; }
+	bool reload() override;
+	bool empty() const override { return false; }
+
+	uint64 size() const override { return 0UL; }
+	uint64 totalSize() const override { return 0UL; }
+	uint64 cacheLimit() const override { return 0UL; }
+	uint32 getOptions() const override { return _options; }
+	void setOptions(uint32 options) override { _options = options; }
+
+private:
+	typedef struct fileIndexEntry
+	{
+		char fname[MAX_PATH];
+		tx_wstring directory;
+		uint32 siz;
+		uint32 fmt;
+	} fileIndexEntry_t;
+
+	bool _createFileIndex(bool update);
+	bool _createFileIndexInDir(tx_wstring directory, bool update);
+
+	uint32 _options;
+	tx_wstring _fullTexPath;
+	tx_wstring _ident;
+	char _identc[MAX_PATH];
+	dispInfoFuncExt _callback;
+	TxHiResCache *_txHiresCache;
+	std::map<uint64, fileIndexEntry_t> _filesIndex;
+	std::map<uint64, GHQTexInfo> _loadedTex;
+};
+
+TxNoCache::TxNoCache(uint32 options, const wchar_t *fullTexPath, const wchar_t *ident, TxHiResCache* txHiResCache, dispInfoFuncExt callback)
+	: _options(options)
+	, _fullTexPath(fullTexPath)
+	, _txHiresCache(txHiResCache)
+	, _ident(ident)
+	, _callback(callback)
+{
+	/* store this for _createFileIndexInDir */
+	wcstombs(_identc, _ident.c_str(), MAX_PATH);
+
+#ifdef OS_WINDOWS
+	/* lowercase on windows */
+	for (uint32 i = 0; i < strlen(_identc); i++) _identc[i] = tolower(_identc[i]);
+#endif
+
+	_createFileIndex(false);
+}
+
+TxNoCache::~TxNoCache()
+{
+	clear();
+}
+
+bool TxNoCache::get(Checksum checksum, GHQTexInfo *info)
+{
+	if (!checksum) {
+		return false;
+	}
+
+	fileIndexEntry_t entry;
+
+#ifdef DEBUG
+	uint32 chksum = checksum._checksum & 0xffffffff;
+	uint32 palchksum = checksum._checksum >> 32;
+#endif
+
+	/* loop over each file from the index and try to match it with checksum */
+	auto indexEntry = _filesIndex.find(checksum);
+	if (indexEntry == _filesIndex.end()) {
+		DBG_INFO(80, wst("TxNoCache::get: chksum:%08X %08X not found\n"), chksum, palchksum);
+		return false;
+	}
+
+	entry = indexEntry->second;
```

Comment:
better 
const fileIndexEntry_t& entry = indexEntry->second;
to avoid copy of the entry.

Reply:
I can't make it const, else I get the following error:
```
/home/rosalie/dev/GLideN64/src/GLideNHQ/TxHiResNoCache.cpp: In member function ‘virtual bool TxHiResNoCache::get(Checksum, GHQTexInfo*)’:
/home/rosalie/dev/GLideN64/src/GLideNHQ/TxHiResNoCache.cpp:101:61: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
  101 |         uint8_t* tex = TxHiResLoader::loadFileInfoTex(entry.fname, entry.siz, &width, &height, entry.fmt, &format);
      |                                                       ~~~~~~^~~~~
      |                                                             |
      |                                                             const char*
```